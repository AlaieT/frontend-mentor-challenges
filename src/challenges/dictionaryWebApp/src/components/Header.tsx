import React from "react";

import FontSelect from "./FontSelect";
import ModeChange from "./ModeChange";

import { ReactComponent as LogoIcon } from "../assets/images/logo.svg";

import type { HeaderProps } from "../types";

import styles from "../styles/components/header.module.scss";

const Header = ({
  font,
  mode,
  handleFont = () => undefined,
  handleMode = () => undefined
}: HeaderProps) => (
  <header id={styles.header}>
    <LogoIcon width={32} height={36.5} />
    <div id={styles.controls}>
      <FontSelect callback={handleFont} font={font} mode={mode} />
      <span id={styles.separate} />
      <ModeChange callback={handleMode} mode={mode} />
    </div>
  </header>
);

export default Header;
