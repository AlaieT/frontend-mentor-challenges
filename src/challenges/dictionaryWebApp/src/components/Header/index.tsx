import React from "react";

import { GlobalContext } from "../../utils/context";

import { ReactComponent as LogoIcon } from "../../assets/images/logo.svg";
import { ReactComponent as MoonIcon } from "../../assets/images/icon-moon.svg";

import styles from "../../styles/components/header.module.scss";

const Header = () => {
  const globalContext = React.useContext(GlobalContext);

  return (
    <header>
      <LogoIcon width={32} height={36.5} />
      <div id={styles.controls}>
        <div id={styles.mode}>
          <MoonIcon width={20} height={20} />
        </div>
      </div>
    </header>
  );
};

export default Header;
