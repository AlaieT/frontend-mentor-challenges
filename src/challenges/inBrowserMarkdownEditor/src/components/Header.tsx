import React, { useRef, useState } from "react";

import { ReactComponent as IconMenu } from "../assets/images/icon-menu.svg";
import { ReactComponent as IconClose } from "../assets/images/icon-close.svg";
import { ReactComponent as IconDelete } from "../assets/images/icon-delete.svg";
import { ReactComponent as IconSave } from "../assets/images/icon-save.svg";
import { ReactComponent as IconDocument } from "../assets/images/icon-document.svg";
import { ReactComponent as Logo } from "../assets/images/logo.svg";
import styles from "../styles/components/header.module.scss";
import { HeaderProps } from "../types";

const Header = ({
  isMenuOpen,
  handleOpenMenu,
  handleDocumentName,
  documentName = "welcome.md",
  handleDelete,
  handleSave
}: HeaderProps) => {
  const nameInputRef = useRef<HTMLInputElement>(null);
  const [isNameChange, setIsNameChange] = useState(false);
  const handleIsNameChange = (
    e:
      | React.MouseEvent<HTMLButtonElement>
      | React.FocusEvent<HTMLInputElement, Element>
  ) => {
    e.preventDefault();
    setIsNameChange((item) => !item);

    if (isNameChange) {
      handleDocumentName(nameInputRef.current?.value || "");
    }
  };

  return (
    <header id={styles.header}>
      <button id={styles["menu-button"]}>
        {isMenuOpen ? <IconClose /> : <IconMenu />}
      </button>
      <div id={styles.title}>
        <Logo />
      </div>
      <span id={styles.split} />
      <div id={styles["document-name"]}>
        <IconDocument/>
        <div id={styles["name-input"]}>
          <p>Document Name</p>
          <div id={styles["input-button-area"]}>
            {isNameChange ? (
              <input
                ref={nameInputRef}
                // eslint-disable-next-line jsx-a11y/no-autofocus
                autoFocus
                defaultValue={documentName}
                onBlur={handleIsNameChange}
              />
            ) : (
              <button onClick={handleIsNameChange}>{documentName}</button>
            )}
          </div>
        </div>
      </div>
      <button id={styles.delete}>
        <IconDelete />
      </button>
      <button id={styles.save}>
        <IconSave width={16} height={16} />
        <p>Save Changes</p>
      </button>
    </header>
  );
};

export default Header;
