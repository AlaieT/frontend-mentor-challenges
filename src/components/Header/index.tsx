import React from "react";

import Title from "../Title";
import Score from "../Score";

import type { HeaderProps } from "../../types";

import styles from "../../styles/components/header.module.scss";

const Header = ({ challengesCount }: HeaderProps) => (
  <header>
    <ul id={styles.credentials}>
      <li>
        <p>Ilya「Alaie」Shafeev</p>
      </li>
      <li>
        <p>PROJECTS GALLERY</p>
      </li>
      <li>
        <p>©2023</p>
      </li>
    </ul>
    <Title sentence="frontend mentor—challenges" delay={50} />
    <ul id={styles.description}>
      <li>
        <p>
          This website contains my solutions for Frontend Mentor challenges.
        </p>
      </li>
      <li>
        <p>
          All challenges made as individual web projects, so they include using
          of frameworks, linters,
        </p>
      </li>
      <li>
        <p>
          testing and e.t.c. to make practice building websites in a realistic
          workflow.
        </p>
      </li>
      <li>
        <p>
          All of these challenges help improve skills and gain experience
          creating websites.
        </p>
      </li>
    </ul>
    <Score score={challengesCount} delay={25} />
    <div id={styles.scroll}>
      <p>scroll down</p>
    </div>
  </header>
);

export default Header;
