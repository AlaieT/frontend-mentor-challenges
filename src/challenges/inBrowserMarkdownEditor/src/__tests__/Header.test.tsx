import React from "react";
import { render, cleanup } from "@testing-library/react";

import Header from "../components/Header";

afterEach(cleanup);

describe("Header", () => {
  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(
          <Header documentName="some-document" />
        ).asFragment()
      ).toMatchSnapshot();
    });
  });
});
