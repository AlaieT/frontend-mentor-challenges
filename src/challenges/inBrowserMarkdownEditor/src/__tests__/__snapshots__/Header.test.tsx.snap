// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header correct render should match snapshot 1`] = `
<DocumentFragment>
  <header
    id="header"
  >
    <button
      id="menu-button"
    >
      <div />
    </button>
    <div
      id="title"
    >
      <div />
    </div>
    <span
      id="split"
    />
    <div
      id="document-name"
    >
      <div />
      <div
        id="name-input"
      >
        <p>
          Document Name
        </p>
        <div
          id="input-button-area"
        >
          <button>
            some-document
          </button>
        </div>
      </div>
    </div>
    <button
      id="delete"
    >
      <div />
    </button>
    <button
      id="save"
    >
      <div
        height="16"
        width="16"
      />
      <p>
        Save Changes
      </p>
    </button>
  </header>
</DocumentFragment>
`;
